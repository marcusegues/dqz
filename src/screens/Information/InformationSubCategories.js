// @flow
import React from 'react';
import { View } from 'react-native';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
import { ScrollViewCard } from '../../components/General Components/ScrollViewCard';
import { InformationRow } from './subComponents/InformationRow';
import {
  informationImages,
  informationNavigateTo,
  informationSubCategories,
} from './InformationData';
import { HeaderTitle } from '../../components/Headers/subcomponents/HeaderTitle';
import type { Navigation, TFunction } from '../../types/generalTypes';

import { MainContentContainer } from '../../components/MainContentContainer/MainContentContainer';
import { CardHeaderText } from '../../components/QuestionAnswer/Cards/subcomponents/CardHeaderText';
import { moderateScale } from '../../styles/Scaling';

class InformationSubCategoriesInner extends React.Component<{
  t: TFunction,
  navigation: Navigation,
}> {
  static navigationOptions = ({ screenProps }) => ({
    headerTitle: (
      <HeaderTitle text={screenProps.t('information:informationTitle')} />
    ),
  });

  render() {
    const { t, navigation } = this.props;
    const { infoCategory } = navigation.state.params;

    return (
      <MainContentContainer>
        <ScrollViewCard>
          <View style={{ margin: moderateScale(16) }}>
            <CardHeaderText text={t('informationSubCategoryTitle')} />
          </View>
          {informationSubCategories[infoCategory].map((cat, idx) => (
            <InformationRow
              key={cat}
              borderTop={idx === 0}
              source={informationImages[cat]}
              mainText={t(`${cat}MainText`)}
              subText={t(`${cat}SubText`)}
              rowOnPress={() => navigation.navigate(informationNavigateTo[cat])}
            />
          ))}
        </ScrollViewCard>
      </MainContentContainer>
    );
  }
}

export const InformationSubCategories = (translate(['information'])(
  InformationSubCategoriesInner
): ComponentType<{}>);
