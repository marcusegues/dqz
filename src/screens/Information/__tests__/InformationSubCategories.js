import React from 'react';
// $FlowFixMe
import { shallow } from 'enzyme';
import { InformationSubCategories } from '../InformationSubCategories';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('InformationSubCategories', () => {
  test('InformationSubCategories renders as expected', () => {
    const wrapper = shallow(<InformationSubCategories />);
    expect(wrapper).toMatchSnapshot();
  });
  test('InformationSubCategories renders as expected with language', () => {
    const wrapper = shallow(
      <InformationSubCategories
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  // test('InformationSubCategories renders as expected with language with dive()', () => {
  //   const wrapper = shallow(
  //     <InformationSubCategories
  //       navigation={{ state: { params: { onPress: clickFn } } }}
  //       i18n={{ language: 'de', t: 'test' }}
  //       infoCategories={categories}
  //     />
  //   );
  //   expect(wrapper.dive()).toMatchSnapshot();
  // });
});
