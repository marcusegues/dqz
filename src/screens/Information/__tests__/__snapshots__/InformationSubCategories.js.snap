// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationSubCategories InformationSubCategories renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InformationSubCategories InformationSubCategories renders as expected with language 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    i18n={
      Object {
        "language": "de",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "i18n": Object {
        "language": "de",
      },
      "navigation": Object {
        "dispatch": [Function],
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "i18n": Object {
          "language": "de",
        },
        "navigation": Object {
          "dispatch": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
